File: documentation-examples-java/src/java/dev/typr/foundations/docs/sqlite/JsonType.java
package dev.typr.foundations.docs.sqlite;

import dev.typr.foundations.SqliteType;
import dev.typr.foundations.SqliteTypes;
import dev.typr.foundations.data.Json;

@SuppressWarnings("unused")
public class JsonType {
  // start
  // Backed by TEXT — use SQLite's built-in JSON1 functions (json(), json_extract(), ->, ->>).
  SqliteType<Json> jsonType = SqliteTypes.json;

  Json data = new Json("{\"name\": \"SQLite\"}");
  // stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/sqlite/JsonType.scala
package dev.typr.foundationssc.docs.sqlite
import dev.typr.foundationssc.*
import dev.typr.foundations.data.Json

@SuppressWarnings(Array("unused"))
object JsonType:
  // start
  val jsonType: SqliteType[Json] = SqliteTypes.json

  val data: Json = new Json("{\"name\": \"SQLite\"}")
  // stop
